src/include/snapper/dialogs.rb

```diff
@@ -1233,6 +1018,196 @@ def ShowDialog
+            HSpacing(65),
+            content,
+            VSpacing(0.5),
+            ButtonBox(
+              PushButton(Id(:ok), Label.OKButton),
+              PushButton(Id(:cancel), Label.CancelButton)
+            ),
+            VSpacing(0.5)
+          ),
+          HSpacing(1)
+        )
+      )
+    end
+
+    def snapshot_term(prefix, data)
+      data = deep_copy(data)
+      HBox(
+        HSpacing(),
+        Frame(
+          "",
+          HBox(
+            HSpacing(0.4),
+            VBox(
+              # text entry label
+              InputField(
+                Id(Ops.add(prefix, "description")),
+                Opt(:hstretch),
+                _("Description"),
+                Ops.get_string(data, "description", "")
+              ),
+              # text entry label
+              InputField(
+                Id(Ops.add(prefix, "userdata")),
+                Opt(:hstretch),
+                _("User data"),
+                Snapper.userdata_to_string(data["userdata"])
+              ),
+              Left(
+                ComboBox(
+                  Id(Ops.add(prefix, "cleanup")),
+                  Opt(:editable, :hstretch),
+                  # combo box label
+                  _("Cleanup algorithm"),
+                  cleanup_items(Ops.get_string(data, "cleanup", ""))
+                )
+              )
+            ),
+            HSpacing(0.4)
+          )
+        ),
+        HSpacing()
+      )
+    end
+
+    def update_snapshots
+      # busy popup message
+      Popup.ShowFeedback("", _("Reading list of snapshots..."))
+
+      Snapper.ReadSnapshots()
+
+      Popup.ClearFeedback
+
+      UI.ChangeWidget(Id(:snapshots_table), :Items, get_snapshot_items)
+      selected = UI.QueryWidget(Id(:snapshots_table), :SelectedItems) || []
+      enable_buttons([:modify, :show, :delete], selected.size == 1)
+      enable_buttons([:delete], selected.size >= 1)
+
+    end
+
+    def get_snapshot_items
+      snapshot_items = []
+
+      Snapper.snapshots.each_with_index do |s,i|
+        num = s["num"] || 0
+        start_date = (num != 0) ? timestring(s["date"]) : ""
+        end_date = ""
+        userdata = Snapper.userdata_to_string(s["userdata"])
+        desc = s["description"].to_s
+        if s["type"] == :SINGLE
+          type = _("Single")
+        elsif s["type"] == :POST
+          pre = s["pre_num"] || 0 # pre canot be 0
+          index = Ops.get(Snapper.id2index, pre, -1)
+          if pre == 0 || index == -1
+            Builtins.y2warning(
+              "something wrong - pre:%1, index:%2",
+              pre,
+              index
+            )
+            next
+          end
+          desc = Ops.get_string(Snapper.snapshots, [index, "description"], "")
+          end_date = start_date
+          start_date = timestring(Snapper.snapshots[index]["date"])
+          num = "%{pre} & %{post}" % { :pre => pre, :post => num }
+          type = _("Pre & Post")
+        else
```

Comment:
in general it is not good habit to trust that there is not more types, as if added new one beside PRE, POST and SINGLE this code will write new one as PRE.
so I think better form is 
```
case s["type"]
when :SINGLE
   ....
when :POST
   ....
when :PRE
   ....
else
  raise "Unknown snapshot type #{s["type"]}"
```